pkg/blob/blob.go

```diff
@@ -115,17 +116,35 @@ func (d *Driver) Run(endpoint, kubeconfig string, testBool bool) {
 	}
 	klog.Infof("\nDRIVER INFORMATION:\n-------------------\n%s\n\nStreaming logs below:", versionMeta)
 
-	cloud, err := GetCloudProvider(kubeconfig)
-	if err != nil || cloud.TenantID == "" || cloud.SubscriptionID == "" {
-		klog.Fatalf("failed to get Azure Cloud Provider, error: %v", err)
-	}
-	d.cloud = cloud
-
+	// For the controller server, the NodeID is nil. For the node server, the NodeID is not nil.
 	if d.NodeID == "" {
+		cloud, err := getCloudProvider(kubeconfig)
+		if err != nil || cloud.TenantID == "" || cloud.SubscriptionID == "" {
+			klog.Fatalf("failed to get Azure Cloud Provider, error: %v", err)
+		}
+		d.cloud = cloud
 		// Disable UseInstanceMetadata for controller to mitigate a timeout issue using IMDS
 		// https://github.com/kubernetes-sigs/azuredisk-csi-driver/issues/168
 		klog.Infoln("disable UseInstanceMetadata for controller")
 		d.cloud.Config.UseInstanceMetadata = false
+		klog.Infoln("Starting the controller server...")
+	} else {
+		cloud, err := getCloudProvider(kubeconfig)
+		if err == nil {
+			d.cloud = cloud
+		} else {
+			klog.V(2).Infof("there is no azure.json provided for node server")
+			d.cloud = &azure.Cloud{}
+			kubeClient, err := getKubeClient(kubeconfig)
```

Comment:
what about move all these logic into `getCloudProvider` func:
```
isController := (d.NodeID == "")
getCloudProvider(kubeconfig, isController)
```

That would be more clear since there is already `kubeClient, err := getKubeClient(kubeconfig)` in `getCloudProvider` func

Reply:
done.